test/unit/StakingProducts/recalculateEffectiveWeight.js

```diff
@@ -0,0 +1,447 @@
+const { increaseTime } = require('../../utils').evm;
+
+const DEFAULT_PRODUCTS = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20];
+describe('recalculateEffectiveWeight', function () {
+  it('should return 0 for products not found in stakingPool', async function () {
+    const { stakingProducts } = this;
+    const productIdToAdd = Zero;
+
+    await setStakedProducts.call(this, { productIds: [productIdToAdd] });
+
+    const stakedProduct = await stakingProducts.getProduct(this.poolId, productIdToAdd);
+    expect(stakedProduct.lastEffectiveWeight).to.be.equal(100);
+    expect(stakedProduct.targetWeight).to.be.equal(100);
+
+    const unknownProduct = await stakingProducts.getProduct(this.poolId, productIdToAdd.add(1));
+    expect(unknownProduct.lastEffectiveWeight).to.be.equal(0);
+
+    expect(await stakingProducts.getTotalTargetWeight(this.poolId)).to.be.equal(100);
+    expect(await stakingProducts.getTotalEffectiveWeight(this.poolId)).to.be.equal(100);
+
+    // recalculating should do nothing
+    await stakingProducts.recalculateEffectiveWeights(this.poolId, [productIdToAdd, productIdToAdd.add(1)]);
+
+    expect(await stakingProducts.getTotalTargetWeight(this.poolId)).to.be.equal(100);
+    expect(await stakingProducts.getTotalEffectiveWeight(this.poolId)).to.be.equal(100);
+
+    {
+      const unknownProduct = await stakingProducts.getProduct(this.poolId, productIdToAdd.add(1));
+      expect(unknownProduct.lastEffectiveWeight).to.be.equal(0);
+      expect(unknownProduct.targetWeight).to.be.equal(0);
+    }
+  });
+
+  // TODO: implement once https://github.com/NexusMutual/smart-contracts/issues/842 is sorted out
+  it.skip('effective weight should be > target when allocations are greater than capacity', async function () {});
+
+  it('should calculate effective weight properly when decreasing target weight', async function () {
+    const { stakingProducts } = this;
+    const [staker, coverBuyer] = this.accounts.members;
+
+    const amount = parseEther('12345');
+    const coverBuyAmount = amount.mul(8).div(100).mul(2); // 8% of capacity with 2x capacity multiplier
+
+    const productIdToAdd = Zero;
+    await setStakedProducts.call(this, { productIds: [productIdToAdd], targetWeight: 10 });
+
+    // deposit stake
+    await depositTo.call(this, { staker, amount });
+
+    // buy cover
+    await allocateCapacity.call(this, { coverBuyer, amount: coverBuyAmount, productId: productIdToAdd });
+
+    // recalculate effective weight
+    await stakingProducts.recalculateEffectiveWeights(this.poolId, [productIdToAdd]);
+
+    expect(await stakingProducts.getTotalTargetWeight(this.poolId)).to.be.equal(10);
+    expect(await stakingProducts.getTotalEffectiveWeight(this.poolId)).to.be.equal(10);
+    {
+      const stakedProduct = await stakingProducts.getProduct(this.poolId, productIdToAdd);
+      expect(stakedProduct.lastEffectiveWeight).to.be.equal(10);
```

Comment:
named constant for the 10 as well

`initialTargetWeight`

Makes reading and refacotring much easier